component/src/main/java/io/siddhi/extension/io/kafka/sink/KafkaSink.java

```diff
@@ -224,10 +243,14 @@ public void publish(Object payload, DynamicOptions dynamicOptions, KafkaSinkStat
 
             if (null == partitionNo) {
                 producer.send(new ProducerRecord<>(topic, null, System.currentTimeMillis(), key,
-                        payloadToSend));
+                        payloadToSend),
+                        new KafkaProducerCallBack(topic, null, System.currentTimeMillis()));
             } else {
-                producer.send(new ProducerRecord<>(topic, Integer.parseInt(partitionNo), System.currentTimeMillis(),
-                        key, payloadToSend));
+                Integer partition = Integer.parseInt(partitionNo);
+                long currentTime = System.currentTimeMillis();
+                producer.send(new ProducerRecord<>(topic, partition, currentTime,
+                        key, payloadToSend),
+                        new KafkaProducerCallBack(topic, partition, currentTime));
             }
```

Comment:
Shall we simplify this?
Something like:
```java
Integer partition = (partitionNo != null) ? Integer.parseInt(partitionNo) : null;
long currentTime = System.currentTimeMillis();
producer.send(new ProducerRecord<>(topic, partition, currentTime,
            key, payloadToSend),
            new KafkaProducerCallBack(topic, partition, currentTime));
```

Reply:
This is executing in the `else` block in which partitionNo is not null.
```
if (null == partitionNo) {
               
} else {
    Integer partition = Integer.parseInt(partitionNo);  //partitionNo is not null since this is executed in the else block
}
```

Reply:
The following line will do the null check as you have mentioned.
```java
Integer partition = (partitionNo != null) ? Integer.parseInt(partitionNo) : null;
```
If `partitionNo` is not null, we'll parse it and get a non null `partition`. Otherwise, we'll get a null `partition`.

Then the following lines would function as same as what the existing code does:
```java
long currentTime = System.currentTimeMillis();
producer.send(
            new ProducerRecord<>(
                  topic, 
                  partition,  // either 'null', or a valid Integer (decided by the above code block)
                  currentTime,
                  key, 
                  payloadToSend),
            new KafkaProducerCallBack(
                  topic, 
                  partition, // either 'null', or a valid Integer (decided by the above code block)
                  currentTime));
```

Reply:
I see. You meant to remove the whole if-else block (That if-else block was already there even before the metrics implementation) 
This is another way of writing that same logic :-) I am not sure what value addition this provides.